apps/courses/views.py

```diff
@@ -320,3 +322,21 @@ def search(request):
             "results_count": len(results),
         },
     )
+
+
+# Crea
+@cache_control(must_revalidate=True)
+def create(request):
+    data = cache.get_or_set(
```

Comment:
Ah, si hay un posible tema con el rendimiento. No me gusta pasar tanta información al template de esta forma, pero tampoco creo que cause problemas. Por lo menos localmente no noté anomalías notorias en el rendimiento.

Reply:
Cuál crees que sería una mejor forma? Creo que esos son necesarios para los filtros, entonces tienen que estar disponibles en la vista.

Un porblema de rendimiento que hay ahí es que lo que _setea_ si no existe `"create_data"` en caché es un diccionario ya generado, el cual llama **a el método `available` 5 veces, realizando 5 consultas cada vez** (en [`Sections`](https://github.com/open-source-uc/ramos-uc/blob/544ed233d5a65fc02e471ea6069018c87e3ccac1/apps/courses/models/section.py#L31-L33) y en [`Courses`](https://github.com/open-source-uc/ramos-uc/blob/451e017822fa5acd16eec58bdd3372157811ff01/apps/courses/models/course.py#L7-L9)). Para que no se realizen cada vez y funcione el caché **debería ser una función**. ([Docs de `get_or_set`](https://docs.djangoproject.com/en/3.2/topics/cache/#django.core.caches.cache.get_or_set))

Reply:
Una vez mergeada esta PR realizo los cambios para arreglar esto, debería ser sencillo.

Reply:
Pude verificar esto, el tiempo de respuesta de planifica pasa de 165ms (actual) a 35ms (con cache arreglado) 🤯

Reply:
Genial!